### DISCUSSION TITLE
Building a Google Docs Clone with Supabase Realtime

### DISCUSSION THREAD

**Discussion by kishansripada** [supabase/supabase#9806] [Questions]
Hi there,

I'm looking to build an app similar to Google Docs in that it is collaborative and updates should reflect in realtime when another user makes changes to a document (row).

Right now, I have a useEffect that listens to changes of the document state and debounces the changes so that changes will only persist after 1 second, preventing unnecessary requests.

I'm wondering the best way to send instructions on how to _**change**_ the json document to the database rather than sending a whole new copy of the data. 

Also i'm not sure how google docs works but I've heard it communicates directly from client to client?

Is there an optimal way to create a google docs clone with Supabase?

Thank you!

**Comment by dshukertjr** [comment]
For client to client communication, you can explore broadcast and presence feature of realtime
https://supabase.com/docs/guides/realtime
